src/main/java/org/opensearch/flowframework/rest/RestCreateWorkflowAction.java

```diff
@@ -253,7 +253,11 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
 
             return channel -> client.execute(CreateWorkflowAction.INSTANCE, workflowRequest, ActionListener.wrap(response -> {
                 XContentBuilder builder = response.toXContent(channel.newBuilder(), ToXContent.EMPTY_PARAMS);
-                channel.sendResponse(new BytesRestResponse(RestStatus.CREATED, builder));
+                if (provision || reprovision) {
+                    channel.sendResponse(new BytesRestResponse(RestStatus.ACCEPTED, builder));
+                } else {
+                    channel.sendResponse(new BytesRestResponse(RestStatus.CREATED, builder));
+                }
```

Comment:
This works but could be simpler inline:
```suggestion
                channel.sendResponse(new BytesRestResponse(provision || reprovision ? RestStatus.ACCEPTED : RestStatus.CREATED, builder));
```